aiida_quantumespresso/workflows/pw/base.py

```diff
@@ -302,6 +298,33 @@ def set_max_seconds(self, max_wallclock_seconds):
         max_seconds = max_wallclock_seconds * max_seconds_factor
         self.ctx.inputs.parameters['CONTROL']['max_seconds'] = max_seconds
 
+    def set_restart_type(self, restart_type, calculation):
+        """Set the restart type for the calculation."""
+
+        if restart_type == RestartType.FROM_SCRATCH:
+            self.ctx.inputs.parameters['CONTROL']['restart_mode'] = 'from_scratch'
+            self.ctx.inputs.parameters['ELECTRONS'].pop('startingpot', None)
```

Comment:
Is the `ELECTRONS` key guaranteed to be set? The `CONTROL` key is explicitly created at the start of the workchain in `validate_parameters`. Maybe we should do the same for `ELECTRONS` (and `SYSTEM` as well while we are at it?) We could then also remove all other occurrences during the workchain of `parameters.setdefault(CARD, {})` and `parameters.get(CARD, {})` as we are guaranteed that it will exist.